awx/main/tasks/callback.py

```diff
@@ -0,0 +1,203 @@
+import json
+import time
+import logging
+from collections import deque
+
+# Django
+from django.utils.timezone import now
+from django.conf import settings
+from django_guid.middleware import GuidMiddleware
+
+
+# AWX
+from awx.main.tasks.jobs import RunInventoryUpdate, RunProjectUpdate
+from awx.main.redact import UriCleaner
+from awx.main.constants import (
+    MINIMAL_EVENTS,
+)
+
+
+logger = logging.getLogger('awx.main.tasks.jobs')
+
+
+class RunnerCallback:
+    def __init__(self):
+        self.runner_callback.parent_workflow_job_id = None
+        self.runner_callback.host_map = {}
+        self.runner_callback.guid = GuidMiddleware.get_guid()
+        self.runner_callback.job_created = None
+        self.runner_callback.recent_event_timings = deque(maxlen=settings.MAX_WEBSOCKET_EVENT_RATE)
+
+    def event_handler(self, event_data):
+        #
+        # ⚠️  D-D-D-DANGER ZONE ⚠️
+        # This method is called once for *every event* emitted by Ansible
+        # Runner as a playbook runs.  That means that changes to the code in
+        # this method are _very_ likely to introduce performance regressions.
+        #
+        # Even if this function is made on average .05s slower, it can have
+        # devastating performance implications for playbooks that emit
+        # tens or hundreds of thousands of events.
+        #
+        # Proceed with caution!
+        #
+        """
+        Ansible runner puts a parent_uuid on each event, no matter what the type.
+        AWX only saves the parent_uuid if the event is for a Job.
+        """
+        # cache end_line locally for RunInventoryUpdate tasks
+        # which generate job events from two 'streams':
+        # ansible-inventory and the awx.main.commands.inventory_import
+        # logger
+        if isinstance(self, RunInventoryUpdate):
+            # Will removing this 'if'statement from here have any performance impact ?
```

Comment:
@AlanCoding In this new file this line ``` if isinstance(self, RunInventoryUpdate):``` would always return false. 
Would it be wise to just remove this line? 
Any thoughts / suggestion on this ?

Reply:
Just like you introduced a class `RunnerCallbackForProjectUpdate`, you can cover this elegantly by introducing `RunnerCallbackForInventoryUpdate`, then move the logic there, and then delete it here.

You could keep it here and then change references in `jobs.py` to `self.runner_callback.end_line`, if that makes it easier to get to a state of functionality.